File: frontend/src/types/source.ts
export interface SourceListItem {
  id: string;
  name: string;
  source_type: string;
  status: string;
  insight_count: number;
}

export interface InsightStats {
  total: number;
  critical_high: number;
  opportunities: number;
  active_sources: number;
}

File: backend/app/schemas/source.py
"""Pydantic schemas for source-oriented responses."""

import uuid

from pydantic import BaseModel


class SourceCreate(BaseModel):
    """Request body for creating a new source."""

    name: str
    source_type: str
    feed_url: str | None = None
    trust_tier: str
    topics: list[str] = []
    status: str = "active"
    config: dict = {}
    region: str = "global"
    target_roles: list[str] = []


class SourceListItem(BaseModel):
    """Source row with aggregated insight count."""

    id: uuid.UUID
    name: str
    source_type: str
    status: str
    insight_count: int
    region: str = "global"
    target_roles: list[str] = []


class InsightStats(BaseModel):
    """Dashboard KPI stats."""

    total: int
    critical_high: int
    opportunities: int
    active_sources: int
